set.mm

```diff
@@ -256608,6 +256617,42 @@ zero ring (at least if its operations are internal binary operations).
       CEFGHQRSUHUFUGDUIJUEUFUGTUEUFUGUADAEUBUCUD $.
   $}
 
+  ${
+    $d B a b c $.  $d .0. a b c $.  $d .x. c $.  $d R a b c $.
+    isdomn4.b $e |- B = ( Base ` R ) $.
+    isdomn4.0 $e |- .0. = ( 0g ` R ) $.
+    isdomn4.x $e |- .x. = ( .r ` R ) $.
+    $( A (nonzero) ring is a domain iff the cancellation law for multiplication
+       holds.  (Contributed by SN, 15-Sep-2024.) $)
+    isdomn4 $p |- ( R e. Domn <-> ( R e. NzRing /\
+                          A. a e. ( B \ { .0. } ) A. b e. B A. c e. B
+                               ( ( a .x. b ) = ( a .x. c ) -> b = c ) ) ) $=
```

Comment:
```suggestion
                          A. a e. ( B \ { .0. } ) A. b e. B
                               ( ( a .x. b ) = 0 -> b = 0 ) ) ) $=
```
I think this is sufficient.

Reply:
It is, but the simpler statement is not what's being proved here. We already have isdomn for showing either direction of the simpler statement holds, but for the forward direction of this statement has not been proven.

Reply:
What I meant is: I think it is sufficient to only prove the result I quoted, both for claiming we have proved the characterization with the righ-cancellation law, and as a "library lemma" for future uses.

Reply:
After proving the simpler version... to be honest I can't judge whether either the simpler version or the more complicated version is useful. So they're now both moved to mathbox: [`14fd186`](https://github.com/metamath/set.mm/pull/4206/commits/14fd1862295e96bacde95cdfafca91e4965dd33f)